mne_bids/write.py

```diff
@@ -898,12 +899,33 @@ def write_raw_bids(raw, bids_basename, output_path, events_data=None,
     if verbose:
         print('Copying data files to %s' % bids_fname)
 
-    if ext not in ALLOWED_EXTENSIONS:
-        raise ValueError('ext must be in %s, got %s'
-                         % (''.join(ALLOWED_EXTENSIONS), ext))
-
+    if ext not in ALLOWED_EXTENSIONS[kind]:
+        if kind == 'meg':
+            if verbose:
+                warn('Converting data files to Functional Image Format (FIF)')
+            n_rawfiles = len(raw.filenames)
+            if n_rawfiles > 1:
+                split_naming = 'bids'
+                raw.save(bids_fname, split_naming=split_naming, overwrite=True)
```

Comment:
these lines are not covered by tests apparently.

You should avoid the code duplication with the block

```
        n_rawfiles = len(raw.filenames)
        if n_rawfiles > 1:
            split_naming = 'bids'
            raw.save(bids_fname, split_naming=split_naming, overwrite=True)
        else:
            # This ensures that single FIF files do not have the part param
            raw.save(bids_fname, overwrite=True, split_naming='neuromag')
```

that you have below. See how you can do this by rethinking the branching logic.